pages/spacedeck/[[...roomId]].js

```diff
@@ -0,0 +1,297 @@
+        });
+        await matrixClient.sendMessage(room, {
+            msgtype: 'm.text',
+            body: link,
+        }).catch(() => {
+            setErrorMessage(t('Something went wrong when trying to save the new sketch link'));
+        });
+
+        return room;
+    }
+
+    const removeLink = async () => {
+        setRemovingLink(true);
+        const remove = await spacedeck.deleteSpaceById(content.body.substring(content.body.lastIndexOf('/') + 1)).catch((e) => console.log(e));
+        if (!remove) {
+            setRemovingLink(false);
+
+            return;
+        }
+        if (!remove.ok) {
+        // @TODO callback function to give user feedback when removing on the server fails
+            setRemovingLink(false);
+
+            return;
+        }
+        await auth.getAuthenticationProvider('matrix').removeSpaceChild(serviceSpaceId, roomId);
+        await matrix.leaveRoom(roomId);
+        router.push('/sketch');
+        setRemovingLink(false);
+    };
+
+    const ActionNewSketch = ({ callbackDone }) => {
+        const [sketchName, setSketchName] = useState('');
+        const [loading, setLoading] = useState(false);
+
+        const createNewSketchRoom = async () => {
+            setLoading(true);
+
+            const create = await spacedeck.createSpace(sketchName);
+            const link = getConfig().publicRuntimeConfig.authProviders.spacedeck.baseUrl + '/spaces/' + create._id;
+            const roomId = await createSketchRoom(link, sketchName);
+            router.push(`/sketch/${roomId}`);
+
+            callbackDone && callbackDone();
+            setLoading(false);
+        };
+
+        return (
+            <Form onSubmit={(e) => { e.preventDefault(); createNewSketchRoom(); }}>
+                <input type="text" placeholder={t('sketch name')} value={sketchName} onChange={(e) => setSketchName(e.target.value)} />
+                <button type="submit" disabled={!sketchName || loading}>{ loading ? <LoadingSpinnerInline inverted /> : t('Create sketch') }</button>
+                { errorMessage && <ErrorMessage>{ errorMessage }</ErrorMessage> }
+            </Form>);
+    };
+
+    const ActionExistingSketch = ({ callbackDone }) => {
+        const [sketchName, setSketchName] = useState('');
+        const [sketchLink, setSketchLink] = useState('');
+        const [validLink, setValidLink] = useState(false);
+        const [loading, setLoading] = useState(false);
+
+        const handleExistingSketch = (e) => {
+            // we check if the link is valid for the service (has the same base url)
+            if (e.target.value.includes(getConfig().publicRuntimeConfig.authProviders.spacedeck.baseUrl)) setValidLink(true);
+            else setValidLink(false);
+            setSketchLink(e.target.value);
+        };
+
+        const handleSubmit = async (e) => {
+            setLoading(true);
+            e.preventDefault();
+            await createSketchRoom(sketchLink, sketchName);
```

Comment:
After this line: Shouldn't we `router.push(...` the user to the added sketch after this was submitted?
We do it for other /sketch boards (see `ActionNewSketch`) and also when adding existing /write pads...